<a href="https://colab.research.google.com/github/anjithsm/DataScience/blob/main/Data_Acquisition_Casestudy.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
 #Load the dataset into a pandas DataFrame.
import pandas as pd

In [2]:
df = pd.read_csv("/content/fifa_players.csv")

In [3]:
#Display the first few rows of the DataFrame to understand its structure
df.head()

name                       full_name birth_date  age  height_cm  \
0      L. Messi  Lionel Andrés Messi Cuccittini  6/24/1987   31     170.18   
1    C. Eriksen    Christian  Dannemann Eriksen  2/14/1992   27     154.94   
2      P. Pogba                      Paul Pogba  3/15/1993   25     190.50   
3    L. Insigne                 Lorenzo Insigne   6/4/1991   27     162.56   
4  K. Koulibaly               Kalidou Koulibaly  6/20/1991   27     187.96   

   weight_kgs  positions nationality  overall_rating  potential  ...  \
0        72.1   CF,RW,ST   Argentina              94         94  ...   
1        76.2  CAM,RM,CM     Denmark              88         89  ...   
2        83.9     CM,CAM      France              88         91  ...   
3        59.0      LW,ST       Italy              88         88  ...   
4        88.9         CB     Senegal              88         91  ...   

   long_shots  aggression interceptions  positioning  vision  penalties  \
0          94          48            22           94      94         75   
1          89          46            56           84      91         67   
2          82          78            64           82      88         82   
3          84          34            26           83      87         61   
4          15          87            88           24      49         33   

  composure  marking standing_tackle  sliding_tackle  
0        96       33              28              26  
1        88       59              57              22  
2        87       63              67              67  
3        83       51              24              22  
4        80       91              88              87  

[5 rows x 51 columns]

In [4]:
#Print the summary statistics of the dataset
summary = df.describe()
summary

age     height_cm    weight_kgs  overall_rating     potential  \
count  17954.000000  17954.000000  17954.000000    17954.000000  17954.000000   
mean      25.565445    174.946921     75.301047       66.240169     71.430935   
std        4.705708     14.029449      7.083684        6.963730      6.131339   
min       17.000000    152.400000     49.900000       47.000000     48.000000   
25%       22.000000    154.940000     69.900000       62.000000     67.000000   
50%       25.000000    175.260000     74.800000       66.000000     71.000000   
75%       29.000000    185.420000     79.800000       71.000000     75.000000   
max       46.000000    205.740000    110.200000       94.000000     95.000000   

         value_euro      wage_euro  international_reputation(1-5)  \
count  1.769900e+04   17708.000000                   17954.000000   
mean   2.479280e+06    9902.134628                       1.111674   
std    5.687014e+06   21995.593750                       0.392168   
min    1.000000e+04    1000.000000                       1.000000   
25%    3.250000e+05    1000.000000                       1.000000   
50%    7.000000e+05    3000.000000                       1.000000   
75%    2.100000e+06    9000.000000                       1.000000   
max    1.105000e+08  565000.000000                       5.000000   

       weak_foot(1-5)  skill_moves(1-5)  ...    long_shots    aggression  \
count    17954.000000      17954.000000  ...  17954.000000  17954.000000   
mean         2.945695          2.361034  ...     46.852456     55.816531   
std          0.663691          0.763223  ...     19.429337     17.394047   
min          1.000000          1.000000  ...      3.000000     11.000000   
25%          3.000000          2.000000  ...     32.000000     44.000000   
50%          3.000000          2.000000  ...     51.000000     59.000000   
75%          3.000000          3.000000  ...     62.000000     69.000000   
max          5.000000          5.000000  ...     94.000000     95.000000   

       interceptions   positioning        vision     penalties     composure  \
count   17954.000000  17954.000000  17954.000000  17954.000000  17954.000000   
mean       46.657959     49.857302     53.406260     48.357302     58.680183   
std        20.754649     19.694311     14.156038     15.810844     11.625541   
min         3.000000      2.000000     10.000000      5.000000     12.000000   
25%        26.000000     38.000000     44.000000     38.000000     51.000000   
50%        52.000000     55.000000     55.000000     49.000000     60.000000   
75%        64.000000     64.000000     64.000000     60.000000     67.000000   
max        92.000000     95.000000     94.000000     92.000000     96.000000   

            marking  standing_tackle  sliding_tackle  
count  17954.000000     17954.000000    17954.000000  
mean      47.162861        47.733040       45.705915  
std       20.037346        21.674973       21.285812  
min        3.000000         2.000000        3.000000  
25%       30.000000        27.000000       24.000000  
50%       52.500000        55.000000       52.000000  
75%       64.000000        66.000000       64.000000  
max       94.000000        93.000000       90.000000  

[8 rows x 42 columns]

### **Q1. What is the size of the dataset? (Number of rows and columns)**

In [5]:
df.shape

(17954, 51)

### **Q2.What are the data types of each column?**

In [6]:
df.dtypes

name                              object
full_name                         object
birth_date                        object
age                                int64
height_cm                        float64
weight_kgs                       float64
positions                         object
nationality                       object
overall_rating                     int64
potential                          int64
value_euro                       float64
wage_euro                        float64
preferred_foot                    object
international_reputation(1-5)      int64
weak_foot(1-5)                     int64
skill_moves(1-5)                   int64
body_type                         object
release_clause_euro              float64
national_team                     object
national_rating                  float64
national_team_position            object
national_jersey_number           float64
crossing                           int64
finishing                          int64
heading_accuracy                   int64
short_passing                      int64
volleys                            int64
dribbling                          int64
curve                              int64
freekick_accuracy                  int64
long_passing                       int64
ball_control                       int64
acceleration                       int64
sprint_speed                       int64
agility                            int64
reactions                          int64
balance                            int64
shot_power                         int64
jumping                            int64
stamina                            int64
strength                           int64
long_shots                         int64
aggression                         int64
interceptions                      int64
positioning                        int64
vision                             int64
penalties                          int64
composure                          int64
marking                            int64
standing_tackle                    int64
sliding_tackle                     int64
dtype: object

### **Q3.Are there any missing values in the dataset? If so, which columns have missing values and how many?**

In [7]:
df.isnull()

name  full_name  birth_date    age  height_cm  weight_kgs  positions  \
0      False      False       False  False      False       False      False   
1      False      False       False  False      False       False      False   
2      False      False       False  False      False       False      False   
3      False      False       False  False      False       False      False   
4      False      False       False  False      False       False      False   
...      ...        ...         ...    ...        ...         ...        ...   
17949  False      False       False  False      False       False      False   
17950  False      False       False  False      False       False      False   
17951  False      False       False  False      False       False      False   
17952  False      False       False  False      False       False      False   
17953  False      False       False  False      False       False      False   

       nationality  overall_rating  potential  ...  long_shots  aggression  \
0            False           False      False  ...       False       False   
1            False           False      False  ...       False       False   
2            False           False      False  ...       False       False   
3            False           False      False  ...       False       False   
4            False           False      False  ...       False       False   
...            ...             ...        ...  ...         ...         ...   
17949        False           False      False  ...       False       False   
17950        False           False      False  ...       False       False   
17951        False           False      False  ...       False       False   
17952        False           False      False  ...       False       False   
17953        False           False      False  ...       False       False   

       interceptions  positioning  vision  penalties  composure  marking  \
0              False        False   False      False      False    False   
1              False        False   False      False      False    False   
2              False        False   False      False      False    False   
3              False        False   False      False      False    False   
4              False        False   False      False      False    False   
...              ...          ...     ...        ...        ...      ...   
17949          False        False   False      False      False    False   
17950          False        False   False      False      False    False   
17951          False        False   False      False      False    False   
17952          False        False   False      False      False    False   
17953          False        False   False      False      False    False   

       standing_tackle  sliding_tackle  
0                False           False  
1                False           False  
2                False           False  
3                False           False  
4                False           False  
...                ...             ...  
17949            False           False  
17950            False           False  
17951            False           False  
17952            False           False  
17953            False           False  

[17954 rows x 51 columns]

In [8]:
missin_values_count = df.isnull().sum()
missin_values_count

name                                 0
full_name                            0
birth_date                           0
age                                  0
height_cm                            0
weight_kgs                           0
positions                            0
nationality                          0
overall_rating                       0
potential                            0
value_euro                         255
wage_euro                          246
preferred_foot                       0
international_reputation(1-5)        0
weak_foot(1-5)                       0
skill_moves(1-5)                     0
body_type                            0
release_clause_euro               1837
national_team                    17097
national_rating                  17097
national_team_position           17097
national_jersey_number           17097
crossing                             0
finishing                            0
heading_accuracy                     0
short_passing                        0
volleys                              0
dribbling                            0
curve                                0
freekick_accuracy                    0
long_passing                         0
ball_control                         0
acceleration                         0
sprint_speed                         0
agility                              0
reactions                            0
balance                              0
shot_power                           0
jumping                              0
stamina                              0
strength                             0
long_shots                           0
aggression                           0
interceptions                        0
positioning                          0
vision                               0
penalties                            0
composure                            0
marking                              0
standing_tackle                      0
sliding_tackle                       0
dtype: int64

### **Q4.What are the unique values in a categorical column of your choice? How many unique values are there?**

In [9]:
unique_value = df['full_name'].unique()
print(unique_value)
df['full_name'].nunique()

['Lionel Andrés Messi Cuccittini' 'Christian  Dannemann Eriksen'
 'Paul Pogba' ... 'Jan Bekkema' 'Abdulrahman Al Yami'
 'José Francisco dos Santos Júnior']


17898

### **Q5.For a numerical column of your choice, calculate the mean, median, and standard deviation.**

In [10]:
mean_median_std = df['overall_rating'].agg(['mean','median','std'])

print(mean_median_std)

mean      66.240169
median    66.000000
std        6.963730
Name: overall_rating, dtype: float64


### **Q6.: Identify any potential outliers in a numerical column of your choice. Explain your approach.**

In [11]:
column = 'ball_control'
Q1 = df[column].quantile(0.25)
Q3 = df[column].quantile(0.75)
IQR = Q3-Q1
lower_bound = Q1-1.5*IQR
upper_bound = Q3+1.5*IQR
outliers = df[(df[column] < lower_bound) | (df[column] > upper_bound)]
outliers

name                            full_name  birth_date  \
0               L. Messi       Lionel Andrés Messi Cuccittini   6/24/1987   
3             L. Insigne                      Lorenzo Insigne    6/4/1991   
11           T. Courtois                     Thibaut Courtois   5/11/1992   
12         M. ter Stegen                Marc-André ter Stegen   4/30/1992   
15             P. Dybala          Paulo Bruno Exequiel Dybala  11/15/1993   
...                  ...                                  ...         ...   
17939          L. Modrić                          Luka Modrić    9/9/1985   
17940          E. Hazard                          Eden Hazard    1/7/1991   
17943          Neymar Jr        Neymar da Silva Santos Junior    2/5/1992   
17944  Cristiano Ronaldo  Cristiano Ronaldo dos Santos Aveiro    2/5/1985   
17951         J. Bekkema                          Jan Bekkema    4/9/1996   

       age  height_cm  weight_kgs positions  nationality  overall_rating  \
0       31     170.18        72.1  CF,RW,ST    Argentina              94   
3       27     162.56        59.0     LW,ST        Italy              88   
11      26     198.12        96.2        GK      Belgium              89   
12      26     187.96        84.8        GK      Germany              89   
15      25     152.40        74.8    CAM,RW    Argentina              89   
...    ...        ...         ...       ...          ...             ...   
17939   33     172.72        66.2        CM      Croatia              91   
17940   28     172.72        76.2     LW,CF      Belgium              91   
17943   27     175.26        68.0    LW,CAM       Brazil              92   
17944   34     187.96        83.0     ST,LW     Portugal              94   
17951   22     185.42        89.8        GK  Netherlands              59   

       potential  ...  long_shots  aggression interceptions  positioning  \
0             94  ...          94          48            22           94   
3             88  ...          84          34            26           83   
11            90  ...          17          23            15           13   
12            92  ...          10          43            22           11   
15            94  ...          88          48            32           84   
...          ...  ...         ...         ...           ...          ...   
17939         91  ...          82          62            83           79   
17940         91  ...          80          54            41           87   
17943         92  ...          82          56            36           89   
17944         94  ...          93          63            29           95   
17951         67  ...           9          27            10            5   

       vision  penalties composure  marking standing_tackle  sliding_tackle  
0          94         75        96       33              28              26  
3          87         61        83       51              24              22  
11         44         27        66       20              18              16  
12         70         25        70       25              13              10  
15         87         86        84       23              20              20  
...       ...        ...       ...      ...             ...             ...  
17939      92         82        90       68              76              73  
17940      89         86        91       34              27              22  
17943      88         81        94       27              24              33  
17944      82         85        95       28              31              23  
17951      25         16        47        9              12              13  

[2073 rows x 51 columns]

### **Q7.Find the correlation between numerical columns and discuss any interesting relationships**

In [12]:
correlation = df['crossing'].corr(df['finishing'])
correlation

0.6641825951455514

### **Q8: Group the data by a categorical column and compute the aggregate statistics (e.g., mean, sum) for a numerical column.**

In [13]:
grouped_df = df.groupby('ball_control')['ball_control'].agg(['mean', 'sum'])
grouped_df

mean  sum
ball_control           
5              5.0    5
8              8.0   16
9              9.0   45
10            10.0  510
11            11.0  682
...            ...  ...
92            92.0  368
93            93.0   93
94            94.0  282
95            95.0  190
96            96.0   96

[90 rows x 2 columns]

### **Q9: Based on your analysis, provide a brief summary of any insights or patterns you discovered in the dataset**

  The dataset is fairly large, with 17,954 records and 51 columns, offering a comprehensive view of player attributes, performance metrics, and physical characteristics